core/merginapi.h

```diff
@@ -610,6 +610,7 @@ class MerginApi: public QObject
     QString mApiRoot;
     LocalProjectsManager &mLocalProjects;
     QString mDataDir; // dir with all projects
+    bool mReloadedNeeded = false; // indicate whether project reload is needed after sync
```

Comment:
Let's move this flag to `TransactionStatus` as there might be several syncs running at the same time. 
We can also call it something like "GpkgSchemaChanged" or w/e related to what actually happened.